pinc/ProofProject.inc

```diff
@@ -0,0 +1,365 @@
+<?php
+declare(strict_types=1);
+
+trait ProofPage
+{
+    private function render_page_data(): array
+    {
+        $round_info_array = [];
+        foreach ($this->round->other_rounds_with_visible_usernames as $other_round_id) {
+            $other_round = get_Round_for_round_id($other_round_id);
+            $username = $this->get_username_for_round($other_round);
+            $round_info_array[] = [$other_round->id, $username];
+        }
+        $page_option = ($this->page_state == $this->round->page_out_state) ? "previous" : "current";
+        return
+            $this->get_page_text_array($page_option) +
+            $this->get_page_state_array() +
+            [
+                "pageNum" => pathinfo($this->page_name, PATHINFO_FILENAME),
+                "roundInfoArray" => $round_info_array,
+                "pagename" => $this->page_name,
+                "image_url" => "{$this->project->url}/{$this->page_name}",
+            ];
+    }
+
+    private function get_username_for_round(object $round): string
+    {
+        $res = DPDatabase::query("
```

Comment:
This needs an `sprintf()` for the `image` parameter to escape it like you do in `get_page_text_array()`.

Reply:
done this